native/explorer/src/encoding.rs

```diff
@@ -467,6 +479,7 @@ pub fn term_from_value<'b>(v: AnyValue, env: Env<'b>) -> Result<Term<'b>, Explor
         AnyValue::Date(v) => encode_date(v, env),
         AnyValue::Time(v) => encode_time(v, env),
         AnyValue::Datetime(v, time_unit, None) => encode_datetime(v, time_unit, env),
+        AnyValue::Duration(v, _time_unit) => Ok(Some(v).encode(env)),
```

Comment:
I actually started down that road before I went with integers for simplicity. I still think that may be the right choice, but I wanted to hear y'all's thoughts first.

In particular, bare integers can't capture the units and so don't survive (de)serialization. E.g. this doesn't work:

```elixir
import Explorer.Series

dt1 = from_list([~N[2023-01-01 00:00:00]])
dt2 = from_list([~N[2023-01-02 00:00:00]])

dt2 |> subtract(dt1) |> to_list |> from_list |> add(dt1)
```

You get

```
** (ArgumentError) cannot invoke Explorer.Series.add/2 with mismatched dtypes: :integer and {:datetime, :millisecond}
```

If we had a custom `%Explorer.Duration{}` or similar, the above would work just fine. The downside is that `Explorer` would need to internally represent durations. That would increase the maintenance burden and (possibly?) make it harder to implement custom backends.

I still think it may be worth it but that's the tradeoff IMO.

Reply:
I think it is fine to keep it as integer, but we need to choose a unique representation, which would most likely be nanoseconds?

Reply:
Clarification: I was saying a `%Explorer.Duration{}` struct might be the way to go despite the overhead. (Sorry about my wording.)

Representing durations as integers in Elixir forces the programmer to always pass along type hints whenever a duration is initially created. That may be cumbersome if you're doing multiple transformations involving a duration column. This is unlike datetime where, once you've created the initial series, you can do:

```elixir
datetime = from_list([1649883642 * 1_000 * 1_000], dtype: {:datetime, :millisecond})
#=> #Explorer.Series<
#=>   Polars[1]
#=>   datetime[μs] [54252-09-24 11:40:00.000000]
#=> >
```

then

```elixir
datetime |> to_list |> from_list
#=> #Explorer.Series<
#=>   Polars[1]
#=>   datetime[μs] [54252-09-24 11:40:00.000000]
```

which preserves the precision without having to re-pass in the `:dtype` arg.

However, it looks like we're already in this situation WRT binary:

```elixir
binary = from_list([<<228, 146, 51>>, "Elixir"], dtype: :binary)
#=> #Explorer.Series<
#=>   Polars[2]
#=>   binary [<<228, 146, 51>>, "Elixir"]
#=> >
```

Here you have that same problem (actually it's even worse because we raise):

```elixir
binary |> to_list |> from_list
#=> ** (ArgumentError) argument error
```

So I think my objection isn't as strong.

If we do stay with integers, then I think normalizing on nanosecond precision is the way to go. I'm gonna assume that's the plan for now.

Reply:
There is an issue in normalizing to nanoseconds: a very large value will overflow 64 bits, which is fine for Elixir, but it will lead issues when converting back to Explorer. So I think you are right, `Explorer.Duration` is best way to go indeed.

How would we implement it? Do we keep two fields? Value and precision? Elixir's Calendar types represent the precision as a number (from 0 to 6) we could do the same here and represent it as a number from 0 to 9, but given it will be different anyway, the precision as one of :millisecond | :microsecond | :nanosecond is fine.

Thoughts?

Reply:
Something to add: in Elixir, the precision is about the _representation_. If we say the datetime is a single digit, then it is a single digit precision. In here, the precision is about low-level storage (i.e. the biggest/smallest number we can represent). Given they are different, let's go with the precision being an atom (the time unit).

Reply:
Yeah I was thinking exactly that:

```elixir
%Explorer.Duration{
  value: integer,
  precision: :millisecond | :microsecond | :nanosecond
}
```

We may find we more fields down the road of course. But right now this struct only needs to hold on to enough information to make the mapping into polars unambiguous.

Reply:
Addressed by: [98a8ff7](https://github.com/elixir-explorer/explorer/pull/683/commits/98a8ff75d1114250d52d319f233c72cbd187d92d)